src/System.Linq/src/System/Linq/ThrowHelper.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Linq
+{
+    internal static class ThrowHelper
+    {
+        internal static void ThrowArgumentNullException(ExceptionArgument argument) => throw CreateArgumentNullException(argument);
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Exception CreateArgumentNullException(ExceptionArgument argument) => new ArgumentNullException(argument.ToString());
+
+        internal static void ThrowArgumentOutOfRangeException(ExceptionArgument argument) => throw CreateArgumentOutOfRangeException(argument);
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static Exception CreateArgumentOutOfRangeException(ExceptionArgument argument) => new ArgumentOutOfRangeException(argument.ToString());
```

Comment:
`argument.ToString()` won't work in ProjectN - reflection against private internal implementation details is disabled in ProjectN.

Reply:
Ah, right, I'll switch to a switch.

Reply:
Is it worth having a separate create exception methods if throw methods are not inlined anyway??